File: api-tools/src/test/scala/com/lightbend/lagom/api/tools/tests/scaladsl/AclService.scala
/*
 * Copyright (C) 2016-2017 Lightbend Inc. <https://www.lightbend.com>
 */
package com.lightbend.lagom.api.tools.tests.scaladsl

import akka.NotUsed
import com.lightbend.lagom.scaladsl.api.Service._
import com.lightbend.lagom.scaladsl.api.transport.Method
import com.lightbend.lagom.scaladsl.api.{ Descriptor, Service, _ }

import scala.concurrent.Future

trait AclService extends Service {

  def getMock(id: String): ServiceCall[NotUsed, NotUsed]

  def addMock: ServiceCall[NotUsed, NotUsed]

  def descriptor: Descriptor =
    named("/aclservice").withCalls(
      restCall(Method.GET, "/scala-mocks/:id", getMock _),
      restCall(Method.POST, "/scala-mocks", addMock)
    ).withAutoAcl(true)

}

class AclServiceImpl extends AclService {
  def getMock(id: String) = ServiceCall { _ => Future.successful(NotUsed) }

  def addMock: ServiceCall[NotUsed, NotUsed] = ServiceCall { _ => Future.successful(NotUsed) }
}

File: api-tools/src/test/java/com/lightbend/lagom/api/tools/tests/javadsl/AclService.java
/*
 * Copyright (C) 2016-2019 Lightbend Inc. <https://www.lightbend.com>
 */
package com.lightbend.lagom.api.tools.tests.javadsl;

import akka.NotUsed;
import com.lightbend.lagom.javadsl.api.Descriptor;
import com.lightbend.lagom.javadsl.api.Service;
import com.lightbend.lagom.javadsl.api.ServiceCall;
import com.lightbend.lagom.javadsl.api.transport.Method;

import static com.lightbend.lagom.javadsl.api.Service.*;

public interface AclService extends Service {

    ServiceCall<NotUsed, NotUsed> getMock(String id);

    ServiceCall<NotUsed, NotUsed> addMock();

    default Descriptor descriptor() {
        return named("/aclservice").withCalls(
            restCall(Method.GET,  "/mocks/:id", this::getMock),
            restCall(Method.POST, "/mocks", this::addMock)
        ).withAutoAcl(true);
    }
}
